src/canvas/graphic.ts

```diff
@@ -314,7 +315,7 @@ function brushText(ctx: CanvasRenderingContext2D, el: TSpan, style: TSpanStylePr
         ctx.textAlign = style.textAlign;
         ctx.textBaseline = style.textBaseline;
         if (ctx.setLineDash) {
-            let lineDash = style.lineDash;
+            let lineDash = style.lineDash = normalizeLineDash(style.lineDash);
             let lineDashOffset = style.lineDashOffset;
```

Comment:
It's better not to modify the original value of `style.lineDash`

Reply:
Tweaked in https://github.com/ecomfe/zrender/pull/628/commits/cf5e31670d4ca914b6f24dc40071e3ad6d43e23f